### PR TITLE
Accept additional options to pass to poetry 

### LINKED ISSUE #59: Question: Can it install poetry from a specific commit?
There is currently an error with poetry on Windows and the fix has not been integrated into released version. 

The current recommendation is to install using:
`curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/install-poetry.py | python - --git https://github.com/serverwentdown/poetry.git@1.1-fix-cache-on-windows`

Would something like this be possible with this action?

### PR DESCRIPTION
**Pull request by mario-bermonti**
Accept additional options from the user that are directly passed to poetry when installing it.

This will allow to install directly from github or use any of the other options accepted by [poetry's installer script](https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py).

This closes #59.

### CODE REVIEW

**Comment by mario-bermonti** [comment]
I added 2 tests, one for each "branch" of the if statement block (version specified vs version unspecified by the user).

I would like your feedback to know if this is the right approach. If it is, I would add this feature to the docs.

Even though the tests passed, I got the following message for the 2 tests added:
`Warning: Unexpected input(s) 'additional-options', valid inputs are ['version', 'virtualenvs-create', 'virtualenvs-in-project', 'virtualenvs-path', 'installer-parallel', 'other-options']`

This is expected until the changes are merged or am I missing something?

File: action.yml [hunk 1]
```diff
@@ -25,6 +25,9 @@ inputs:
     description: "Whether to install many packages at once or one by one. This can fix PyPI DNS resolution errors, but also slows down the installation"
     required: false
     default: "true"
+  other-options:
+    description: "Other options to pass directly to poetry. For example --force."
+    required: false
```
Comment (sondrelg) [inline action.yml:30]: Let's call this `installation-arguments` to be as precise as possible.
```suggestion
  installation-arguments:
    description: "Arguments passed directly to the Poetry installation script. For example --force."
    required: false
```

File: action.yml [hunk 2]
```diff
@@ -43,9 +46,9 @@ runs:
         echo -e "\033[33mInstalling Poetry 👷\033[0m\n"
 
         if [ "${{ inputs.version }}" == "latest" ]; then
-          POETRY_HOME=$path python3 $installation_script --yes
+          POETRY_HOME=$path python3 $installation_script --yes ${{ inputs.other_options}}
```
Comment (sondrelg) [inline action.yml:49]:
```suggestion
          POETRY_HOME=$path python3 $installation_script --yes ${{ inputs.installation-arguments}}
```

File: action.yml [hunk 3]
```diff
@@ -43,9 +46,9 @@ runs:
...
         else
-          POETRY_HOME=$path python3 $installation_script --yes --version=${{ inputs.version }}
+          POETRY_HOME=$path python3 $installation_script --yes --version=${{ inputs.version }} ${{ inputs.other_options}}
```
Comment (sondrelg) [inline action.yml:51]:
```suggestion
          POETRY_HOME=$path python3 $installation_script --yes --version=${{ inputs.version }} ${{ inputs.installation-arguments}}
```

File: .github/workflows/test.yml [hunk 4]
```diff
@@ -180,3 +180,29 @@ jobs:
...
+          assert_in "1.1.11"                            "$(poetry --version)"
+
+  test-additional-options-version-specified:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./
+        with:
+          version: 1.1.7
+          additional-options: --force
```
Comment (sondrelg) [inline .github/workflows/test.yml:205]:
```suggestion
          installation-arguments: --force
```

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -180,3 +180,29 @@ jobs:
...
+  test-additional-options-version-unspecified:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./
+        # This test will need to change every time poetry releases a new version
+        # If you're submitting a PR and this fails because Poetry release a new version
+        # feel free to update it
+        with:
+          additional-options: --git https://github.com/python-poetry/poetry.git@master
```
Comment (sondrelg) [inline .github/workflows/test.yml:193]:
```suggestion
          installation-arguments: --git https://github.com/python-poetry/poetry.git@master
```

**Review by sondrelg — COMMENTED** [review]
The warning output you're seeing is likely due to the fact that you've specified `additional-options` in your tests, but called it `other-options` in the `action.yml`. You also passed in `other_options` which I think is incorrect. I think this issue should be resolved by the suggestions I've added.

Otherwise good job, I think this is the way to go! 

In terms of finalizing the PR now, I have one request: would you mind fixing the tests and making them fail *before* applying the suggestions I added? The fact that tests are currently passing indicates to me that we're not actually testing anything useful yet. Ideally we want tests to fail now, then when we fix the argument errors the tests should pass. Does that make sense?

File: .github/workflows/test.yml [hunk 6]
```diff
@@ -180,3 +180,29 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./
+        with:
+          version: 1.1.7
+          additional-options: --force
+      - run: |
+          source .github/scripts/assert.sh
+          assert_in "1.1.7"                            "$(poetry --version)"
```
Comment (sondrelg) [inline .github/workflows/test.yml:208]: I think the easiest way to make the tests fail here is to use the `--git` argument in both tests to install a specific version, then check the version installed. For the test to have value though, we cannot specify 1.1.7 and check that 1.1.7 is installed.

Comment (mario-bermonti) [reply .github/workflows/test.yml:208]: It makes complete sense, @sondrelg. Sorry for the naming mistakes. That was sloppy of me. 

I checked and the —git argument takes precedence over —version in the poetry installer. We can pass both and make sure it installs the version that corresponds to the git hash specified with the —git command. 

I’m going to work in this soon and report back.

File: .github/workflows/test.yml [hunk 7]
```diff
@@ -180,3 +180,29 @@ jobs:
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./
+        with:
+          version: 1.1.7
+          additional-options: --force
+      - run: |
+          source .github/scripts/assert.sh
+          assert_in "1.1.9"                            "$(poetry --version)"
```
Comment (sondrelg) [inline .github/workflows/test.yml:208]:
```suggestion
          assert_in "1.1.9" "$(poetry --version)"
```

File: .github/workflows/test.yml [hunk 8]
```diff
@@ -180,3 +180,29 @@ jobs:
...
+      - uses: actions/checkout@v2
+      - uses: ./
+        # This test will need to change every time poetry releases a new version
+        # If you're submitting a PR and this fails because Poetry release a new version
+        # feel free to update it
+        with:
+          additional-options: --git https://github.com/python-poetry/poetry.git@master
+      - run: |
+          source .github/scripts/assert.sh
+          assert_in "1.1.9"                            "$(poetry --version)"
```
Comment (sondrelg) [inline .github/workflows/test.yml:196]:
```suggestion
          assert_in "1.1.9" "$(poetry --version)"
```

**Review by sondrelg — COMMENTED** [review]
One more thing btw: would it make sense to cut this whitespace for these tests? The whitespace for the final test seems OK, but here it's a little confusing 🙂

File: .github/workflows/test.yml [hunk 9]
```diff
@@ -180,3 +180,29 @@ jobs:
...
           assert_in  "false"                            "$(poetry config installer.parallel)"
+
+  test-additional-options-version-unspecified:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ./
+        # This test will need to change every time poetry releases a new version
+        # If you're submitting a PR and this fails because Poetry release a new version
+        # feel free to update it
```
Comment (sondrelg) [inline .github/workflows/test.yml:191]: Is this still true?

**Review by sondrelg — COMMENTED** [review]
Think we might be able to remove these comments now, but otherwise looks very good!

If tests pass I'll release a new version later tonight 🎉

File: .github/workflows/test.yml (hunk 9 above)
Comment (mario-bermonti) [reply .github/workflows/test.yml]: Not anymore. Cleaned!

**Comment by mario-bermonti** [comment]
I had to submit these changes that just continue to rename the additional-options to installation-arguments in a different commit (dded203) because I made changes to these lines and the suggestions were flagged as outdated.

**Comment by mario-bermonti** [comment]
I am going to need guidance with the documentation. How would you suggest this feature be documented, @sondrelg?

**Comment by sondrelg** [comment]
I think we could add the option in the first two code blocks in the Readme, then add a sentence and an example about how to specify installer arguments at the end of the 'Defaults' section - just below (or above if you think it's better) the stuff about experimental settings 👏

**Comment by sondrelg** [comment]
I pushed final small edit to the readme, but really happy with this feature @mario-bermonti. Thanks a lot 👏

**Comment by mario-bermonti** [comment]
Thank you for a great experience, @sondrelg.

**Pull request merged**